Elaborated documentation regarding BasicAuth (#11)
grafana/cortex-rules-action: README.md

| ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | -------- | ------- |
| `CORTEX_ADDRESS`   | URL address for the target Cortex cluster                                                                                                                                                                                                  | `false`  | N/A     |

> | `CORTEX_TENANT_ID` | ID for the desired tenant in the target Cortex cluster                                                                                                                                                                                     | `false`  | N/A     |

Reviewer: ```suggestion
| `CORTEX_TENANT_ID` | ID for the desired tenant in the target Cortex cluster. Used as the username under HTTP Basic authentication.                                                                                                                                                                                     | `false`  | N/A     |
```

> | `CORTEX_API_KEY`   | Optional password that is required for password protected Cortex clusters. An encrypted [github secret](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets ) is recommended. Will use the `CORTEX_TENANT_ID` as a username | `false`  | N/A     |

Reviewer: ```suggestion
| `CORTEX_API_KEY`   | Optional password that is required for password-protected Cortex clusters. An encrypted [github secret](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets ) is recommended. Used as the password under HTTP Basic authentication. | `false`  | N/A     |
```

| `ACTION`           | Which action to take. One of `lint`, `prepare`, `check`, `diff` or `sync`                                                                                                                                                                  | `true`   | N/A     |
| `RULES_DIR`        | Comma-separated list of directories to walk in order to source rules files                                                                                                                                                                 | `false`  | `./`    |

## Authentication

> Currently cortextool only support basic authentication against a Cortex cluster. To enable basic authentication on this action ensure that a user with the `CORTEX_TENANT_ID` variable username exists on the basic auth proxy and set the `CORTEX_API_KEY` as the password.

Reviewer: ```suggestion
This GitHub Action uses [`cortextool`](https://github.com/grafana/cortex-tools) under the hood, `cortextool` uses HTTP Basic authentication against a Cortex cluster. The variable `CORTEX_TENANT_ID` is used as the username and `CORTEX_API_KEY` as the password.
```
